src/palette-window.cpp

```diff
@@ -1,6 +1,10 @@
 #include <queue>
 #include <utility>
 
+#pragma warning(push, 0)
+#include <FL/platform.H>
+#pragma warning(pop)

```

Comment:
I'm not sure these `warning(push, 0)`/`warning(pop)` are actually useful anymore. What do you think?

Reply:
@Rangi42 So FLTK (1.3 and 1.4) can produce warnings for "conversion from/to, possible loss of data" in several FL header files.
They are mostly for assigning from (int) enum to uchar.

I figured out these warnings are only triggered with `<WarningLevel>Level4</WarningLevel>` in the .vcxproj file.

Level3 (which is what I've always used in crystal-tracker) doesn't produce a warning for these enum/uchar assignments.

I personally like Level3 more than Level4 in general, because I don't really care to be warned about "unreferenced formal parameter" etc.

So the options are:

* Leave the #pragmas
* Switch to Level3
* Get FLTK to add the appropriate casts in the public header files (I'll try to make this happen)

Reply:
https://github.com/fltk/fltk/pull/1126

Reply:
It didn't make it into 1.4.0 but that's okay. I'll just leave the #pragmas.